File: IzquierdoAndres_Musica/IzquierdoAndres_Musica/Models/Artist.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace IzquierdoAndres_Musica.Models;

public partial class Artist
{
    public int ArtistId { get; set; }

    [DisplayName("Nombre")]
    [Required(ErrorMessage = "Obligatorio")]
    [MaxLength(50, ErrorMessage = "El nombre no puede tener más de 50 carácteres.")]
    [MinLength(1, ErrorMessage = "El nombre no puede tener menos de 1 carácter.")]
    public string? Name { get; set; }

    public virtual ICollection<Album> Albums { get; set; } = new List<Album>();
}

File: IzquierdoAndres_SPRING/src/main/java/es/cifpcm/IzquierdoAndres_SPRING/Models/Artist.java
package es.cifpcm.IzquierdoAndres_SPRING.Models;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

import java.io.Serializable;
import java.util.List;

@Entity
@Table(name = "artist")
public class Artist implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @Column(name = "ArtistId", nullable = false)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer artistId;

    @NotBlank(message = "El nombre no puede estar vacío.")
    @Size(min = 1, max = 120, message = "El título debe tener al menos un carácter y como máximo 120.")
    @Column(name = "Name")
    private String name;

    @OneToMany(mappedBy = "artist")
    private List<Album> albums;

    public Artist(Integer artistId, String name, List<Album> albums) {
        this.artistId = artistId;
        this.name = name;
        this.albums = albums;
    }

    public Artist() {

    }

    public Integer getArtistId() {
        return artistId;
    }

    public void setArtistId(Integer artistId) {
        this.artistId = artistId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<Album> getAlbums() {
        return albums;
    }

    public void setAlbums(List<Album> albums) {
        this.albums = albums;
    }
}
